### PR TITLE
Adds XL machine spec and notes on large deployments

### PR DESCRIPTION
**Pull request by topfunky**
- Mentions XL machine specs for 3 cloud vendors
- Links to gossip tuning configuration docs
- Links to non-voting server feature
- Links to performance tuning section of guide

### CODE REVIEW

File: website/source/docs/guides/deployment.html.md [hunk 1]
```diff
@@ -26,11 +26,18 @@ The following instance configurations are recommended.
...
+|XL|12-24 core|64+ GB RAM|SSD|**AWS**: m5d.4xlarge|
+|||||**Azure**: Standard\_D16\_v3, Standard\_D32\_v3|
+|||||**GCE**: n1-standard-32, n1-standard-64|
 
 The **small size** instance configuration is appropriate for most initial production deployments, or for development/testing environments. The large size is for production environments where there is a consistently high workload. Suggested instance types are provided for common platforms, but do refer to platform documentation for up-to-date instance types that align with the recommended resources.
 
 ~> **NOTE** For large workloads, ensure that the disks support a high number of IOPS to keep up with the rapid Raft log update rate.
 
+The recommended size with the default configuration for a single datacenter is 5,000 nodes. Consul can support larger single datacenter cluster sizes by tuning the [gossip parameters](/docs/agent/options.html#gossip_lan) and ensuring Consul agents -- particularly servers -- are running on sufficient hardware. There are real production users of Consul running with greater than 25,000 nodes in a single datacenter today by tuning these parameters.
+
```
Comment (pavbha04) [inline website/source/docs/guides/deployment.html.md]: Looks great. I think we could add to this " There are real production users of Consul running with greater than 25,000 nodes in a single datacenter today by tuning these parameters. The minimum hardware requirement must be XL (Link to above)  or better to achieve such scale.

File: website/source/docs/guides/deployment.html.md [hunk 2]
```diff
@@ -26,11 +26,18 @@ The following instance configurations are recommended.
...
+For a write-heavy and/or a read-heavy cluster, the number of clients may need to be reduced further with considerations for the impact of the number of services and/or watches registered and the number and size of KV pairs. Alternately, large scale read requests can be achieved by increasing the number of non-voting servers ([Enterprise feature](/docs/enterprise/read-scale/index.html)) while maintaining the recommended number of servers (3 or 5) in the quorum. See [Performance Tuning](#1-3-performance-tuning) for more details.
+
```
Comment (pavbha04) [inline website/source/docs/guides/deployment.html.md]: See [Performance Tuning](#1-3-performance-tuning) for more recommendations for ready-heavy clusters

File: website/source/docs/guides/deployment.html.md [hunk 3]
```diff
@@ -26,11 +26,18 @@ The following instance configurations are recommended.
...
+|||||**Azure**: Standard\_D16\_v3, Standard\_D32\_v3|
+|||||**GCE**: n1-standard-32, n1-standard-64|
 
 The **small size** instance configuration is appropriate for most initial production deployments, or for development/testing environments. The large size is for production environments where there is a consistently high workload. Suggested instance types are provided for common platforms, but do refer to platform documentation for up-to-date instance types that align with the recommended resources.
 
 ~> **NOTE** For large workloads, ensure that the disks support a high number of IOPS to keep up with the rapid Raft log update rate.
 
+The recommended size with the default configuration for a single datacenter is 5,000 nodes. Consul can support larger single datacenter cluster sizes by tuning the [gossip parameters](/docs/agent/options.html#gossip_lan) and ensuring Consul agents -- particularly servers -- are running on sufficient hardware. There are real production users of Consul running with greater than 25,000 nodes in a single datacenter today by tuning these parameters. [XL server instances](#system-requirements) or better are required to achieve this scale.
```
Comment (pearkes) [inline website/source/docs/guides/deployment.html.md]: I feel like these two paragraphs fit better in the "Single Datacenter" section of the guide. We have the 5k number there, and we could add this content onto that which is where I would expect to see it more.

Comment (topfunky) [reply website/source/docs/guides/deployment.html.md]: Addressed as requested at #7293971

**Pull request merged**